core/src/sql/value/get.rs

```diff
@@ -36,6 +39,70 @@ impl Value {
 			return Err(Error::ComputationDepthExceeded);
 		}
 		match path.first() {
+			// The knowledge of the current value is not relevant to Part::Recurse
+			Some(Part::Recurse(recurse, inner_path)) => {
+				// Find the path to recurse and what path to process after the recursion is finished
+				let (path, after) = match inner_path {
+					Some(p) => (p.0.as_slice(), path.next().to_vec()),
+					_ => (path.next(), vec![]),
+				};
+
+				// We first try to split out a root-level repeat-recurse symbol
+				// By doing so, we can eliminate un-needed recursion, as we can
+				// simply loop.
+				let (path, plan, after) = match path.split_by_repeat_recurse() {
+					Some((path, local_after)) => (path, None, [local_after, &after].concat()),
+
+					// If we do not find a root-level repeat-recurse symbol, we
+					// can scan for a nested one. We only ever allow for a single
+					// repeat recurse symbol, hence the separate check.
+					_ => match path.find_recursion_plan() {
+						Some((path, plan, local_after)) => {
+							(path, Some(plan), [local_after, &after].concat())
+						}
+						_ => (path, None, after),
+					},
+				};
+
+				// Collect the min & max for the recursion context
+				let (min, max) = recurse.try_into()?;
+				// Construct the recursion context
+				let rec = Recursion {
+					min,
+					max,
+					iterated: &0,
+					current: self,
+					path,
+					plan: plan.as_ref(),
+				};
+
+				// Compute the recursion
+				let v = compute_idiom_recursion(stk, ctx, opt, doc, rec).await?;
+
+				// If we have a leftover path, process it
+				if !after.is_empty() {
+					stk.run(|stk| v.get(stk, ctx, opt, doc, after.as_slice())).await
+				} else {
+					Ok(v)
+				}
+			}
+			// We only support repeat recurse symbol in certain scenarios, to
+			// ensure we can process them efficiently. When encountering a
+			// recursion part, it will find the repeat recurse part and handle
+			// it. If we find one in any unsupported scenario, we throw an error.
```

Comment:
This should probably also be handled in the parser. But that might be best left to a future PR.

Reply:
Let's do this in a future PR yeah, not sure how this would be best handled in the parser, you would need to track context right?